### DISCUSSION TITLE
How to use environment variables in .env file in _quarto.yml file?

### DISCUSSION THREAD

**Discussion by jskherman** [quarto-dev/quarto-cli#2531] [Q&A]
Hi! May I ask on how one would go about using the values in a `.env` file?

I would like to avoid publishing some secrets/credentials on the public repo and using something like GitHub Actions since we publish using `quarto publish netlify` directly (convenient for images).

I've tried poring through the documentation (Guide or Reference) for some hint on how to do this such as `{{< env EXAMPLE_VAR >}}` but I can't seem to find any.

I currently make do with finding and replacing specific strings from the `_quarto.yml` file with a pre-render script and reversing the changes with a post-render script.

Any ideas on how to do this in a less scrappy/native to Quarto way? Thank you.

**Comment by cderv** [comment]
We have added some new feature in 1.2 currently [in pre-release version](https://quarto.org/docs/download/) for Quarto projects:

* Environment variables management: https://quarto.org/docs/projects/environment.html
* Project profiles: https://quarto.org/docs/projects/profiles.html

Using both of those options should be the way to go to manage local vs deployment environment.

**Reply by jskherman** [reply]
Thank you so much for the information. Sorry if this is redundant but after reading through these docs, I still cannot seem to figure out how to use an environment variable inside of the `_quarto.yml` file. I have already tried using an `_environment` and `_environment.local` file with the preview in the pre-release version but no luck here as well.

For example, how do I get the environment variable `SECRET` in an example `.env` or `_environment.local` like so:
```shell
SECRET=VeryImportantSecretNoOneMustKnow
```
and use it inside of `_quarto.yml`? Like so:
```yaml
website:
    title: ${SECRET}
```

**Reply by cderv** [reply]
Oh I see. 

We have a shortcodes available to read some environment variable inside your document: https://quarto.org/docs/authoring/variables.html#env

I don't think it is available to be used inside any YAML fields of `_quarto.yml` (mostly because they could be handle by Pandoc directly) 

@dragonstyle do you know our status on this ? 

However @jskherman for this specific usage 

> use it inside of _quarto.yml

I believe the [Project profiles](quarto.org/docs/projects/profiles.html) are better fit. You could have a project profile that you don't share, and that you are using when rendering in closed environment, to override the default title with your secret title

**Reply by jskherman** [reply]
I see, I'll have a look into the Project Profiles if I can make it work. Thanks!

**Reply by cderv** [reply]
please do share back you experience. The two features mentioned should be the solution to your use case, but maybe we missed something and it is not yet ideal.
